packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -570,20 +662,32 @@ class _CupertinoNavigationBarState extends State<CupertinoNavigationBar> {
 
   @override
   Widget build(BuildContext context) {
-    final Color backgroundColor =
-      CupertinoDynamicColor.maybeResolve(widget.backgroundColor, context) ?? CupertinoTheme.of(context).barBackgroundColor;
+    // The static navigation bar does not expand or collapse (see CupertinoSliverNavigationBar),
+    // it will either display the collapsed nav bar with middle, or the expanded with largeTitle.
+    assert(widget.middle == null || widget.largeTitle == null);
+
+    final Color backgroundColor = CupertinoDynamicColor.maybeResolve(
+      widget.backgroundColor,
+      context,
+    ) ?? CupertinoTheme.of(context).barBackgroundColor;
 
     final Color? parentPageScaffoldBackgroundColor = CupertinoPageScaffoldBackgroundColor.maybeOf(context);
 
     final Border? initialBorder = widget.automaticBackgroundVisibility && parentPageScaffoldBackgroundColor != null
       ? _kTransparentNavBarBorder
       : widget.border;
-    final Border? effectiveBorder = widget.border == null ? null : Border.lerp(initialBorder, widget.border, _scrollAnimationValue,);
+    final Border? effectiveBorder = widget.border == null
+      ? null
+      : Border.lerp(initialBorder, widget.border, _scrollAnimationValue,);
```

Comment:
formatting nit: should there be a trailing comma if there isn't a newline after?